src/global/signal_handler.cc

```diff
@@ -305,7 +305,13 @@ static void handle_fatal_signal(int signum)
     }
   }
 
-  reraise_fatal(signum);
+  if (g_eio) {
+    // if this was an EIO crash, we don't need to trigger a core dump,
+    // since the problem is hardware, or some layer beneath us.
+    _exit(EIO);
```

Comment:
I think this should at least be -EIO, but I'm unsure about the value of an exit code that's an error code.  There seem to be a lot of conventions around.  But a negative number can be confused with a signal exit.

Reply:
I went with a positive value because the "negative" ones are interpreted as signals, and this isn't that.  I figured exit code 5 == EIO would be more meaningful than, say, 1.

Reply:
ah, no, you're right, it's the errno back from syscalls that's negative.  misremembered.